### PR TITLE
api: access-control: allowedOrigins

### PR DESCRIPTION
**Pull request by gioelecerati**
**What does this pull request do? Explain your changes. (required)**

**Specific updates (required)**

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional)**

**Checklist**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/controllers/access-control.test.ts [hunk 1]
```diff
@@ -242,6 +242,48 @@ describe("controllers/access-control", () => {
...
+        stream: `video+${gatedAsset.playbackId}`,
+        type: "jwt",
+        pub: "notExistingPubKey",
+        webhookPayload: {
+          headers: {
+            origin: "http://localhost:3000",
+          },
+        },
+      });
+      expect(res4.status).toBe(403);
```
Comment (victorges) [inline packages/api/src/controllers/access-control.test.ts:280]: Could you add one example that passes all checks as well?

File: packages/api/src/controllers/access-control.ts [hunk 2]
```diff
@@ -242,6 +243,24 @@ app.post(
       return res.end();
     }
 
+    let origin: string = null;
+    if (req.body?.webhookPayload?.headers) {
+      origin = req.body?.webhookPayload?.headers["origin"];
+    }
```
Comment (victorges) [inline packages/api/src/controllers/access-control.ts]: nit: could do in one line if you want
```suggestion
    const origin = req.body?.webhookPayload?.headers?.origin;
```

File: packages/api/src/controllers/asset.ts [hunk 3]
```diff
@@ -287,6 +287,24 @@ async function validateAssetPlaybackPolicy(
...
     }
+    const allowedOrigins = playbackPolicy?.allowedOrigins;
+    if (allowedOrigins) {
+      if (allowedOrigins.length > 0) {
+        const allowedOriginsValid = allowedOrigins.every((origin) => {
+          const url = new URL(origin);
+          return (
+            ["http:", "https:"].includes(url.protocol) &&
+            url.hostname &&
+            url.port
```
Comment (victorges) [inline packages/api/src/controllers/asset.ts]: Not sure if necessary (or even correct), but should we check that there is no path or the path is just `/`? I'm not sure if there could be a path tho, gotta read spec

File: packages/api/src/controllers/asset.ts [hunk 4]
```diff
@@ -287,6 +287,24 @@ async function validateAssetPlaybackPolicy(
         `webhook ${playbackPolicy.webhookId} not found`
       );
     }
+    const allowedOrigins = playbackPolicy?.allowedOrigins;
+    if (allowedOrigins) {
+      if (allowedOrigins.length > 0) {
+        const allowedOriginsValid = allowedOrigins.every((origin) => {
+          const url = new URL(origin);
```
Comment (victorges) [inline packages/api/src/controllers/asset.ts]: You might need to try catch this to throw a friendly error to the user. Or does it never explode here? Maybe this present is very lenient to invalid input, gpt told me that once lol

**Review by victorges — APPROVED** [review]
LGTM

File: packages/api/src/controllers/asset.ts (hunk 3 above)
Comment (gioelecerati) [reply packages/api/src/controllers/asset.ts]: They should never put the path, even with the final slash, I will add a check

**Pull request merged**